### DISCUSSION TITLE
Is it sensible to merge values from localStorage right in the state() method?

### DISCUSSION THREAD

**Discussion by ysrn** [vuejs/pinia#821] [Help and Questions] [answered]
Something like this basically:
```js
export const useStore = defineStore('store', {
	state() {
		const def = {
			someVal: 2,
			anotherVal: {}
		};
		if (localStorage.getItem('store')) {
			Object.assign(def, JSON.parse(localStorage.getItem('store')));
		}
		return def;
	}
});
```
Any potential issues here?

While I'm here, any reason not to use and subscribe to the store right after its definition, for localStorage forwarding?

Something like this right after defining the store:
```js
const store = useStore();
store.$subscribe((mutation, state) => localStorage.setItem('store', JSON.stringify(state)));
```

**Comment by ysrn** [comment]
Just trying to make sure I'm not setting myself up for a future headache here... @posva?

**Comment by posva** [comment, marked as answer]
I would say that, in general, merging everything from a localstorage value to the state will be more trouble than benefit. I recommend you to instead use https://vueuse.org/core/uselocalstorage/#uselocalstorage:

```js
defineStore('id', {
  state: () => ({
    someSavedVariable: useLocalStorage(key, initialValue)
    // other
  })
})

// setup store
defineStore('id', () => {
  const someSavedVariable = useLocalStorage(...)
  // ...
})
```

I would recommend you to share more about why you need to use local storage and people will more likely to be able to help you on this one 🙂 
The Discord chat should also be a nice place to ask this kind of question as it might go a bit beyond pinia itself
